test/scheduler.jl

```diff
@@ -66,31 +66,96 @@ end
     end
     @everywhere (pop!(Dagger.PROCESSOR_CALLBACKS); empty!(Dagger.OSPROC_CACHE))
 
-    @testset "Add new workers" begin
-        using Distributed
-        ps1 = addprocs(2, exeflags="--project");
-    
-        @everywhere begin
+    @testset "Modify workers in running job" begin
+        # Test that we can add/remove workers while scheduler is running.
+        # As this requires asynchronity a Condition is used to stall the tasks to 
+        # ensure workers are actually modified while the scheduler is working 
+
+        setup = quote
            using Dagger, Distributed
            # Condition to guarantee that processing is not completed before we add new workers
+           # Note: c is used in expressions below
            c = Condition()
            function testfun(i)
-               i < 2 && return myid()
+               i < 4 && return myid()
                wait(c)
                return myid()
            end
         end
-    
-        ts = delayed(vcat)((delayed(testfun)(i) for i in 1:4)...);
-        job = @async collect(Context(ps1), ts);
-    
-        ps2 = addprocs(2, exeflags="--project");
-    
-        while !istaskdone(job)
-            @everywhere ps1 notify(c)
+
+        @testset "Add new workers" begin
+        ps = []
+        try     
+                ps1 = addprocs(2, exeflags="--project");
+                push!(ps, ps1)
```

Comment:
I think you mean `append!`, otherwise you have a `Vector{Vector{Int}}`. Apparently passing a vector of the wrong type to `Distributed.rmprocs` silently fails, which is a terrible thing...

Reply:
yikes!